design/gc-for-text-type.md

```diff
@@ -0,0 +1,37 @@
+---
+title: gc-for-text-type
+target-version: 0.1.1
+---
+
+# Garbage collection for Text Type
+
+## Summary
+
+Text and RichText use `RgaTreeSplit`. And when the text is removed, it marks a tombstone on the node corresponding to the text.  
+We provide garbage collection to clean up nodes of text type data that have been removed.
+
+### Goals
+
+When all synchronization between clients is complete, the node marked tombstone is no longer needed.
+As the number of nodes marked tombstone increases, unnecessary memory waste occurs.
+So it provides garbage collection to solve the problem of wasting memory.
+
+### Non-Goals
+
+1. It does not directly control the garbage collection provided by the programming language. It simply removes the relationship of the nodes connected to the linked list and makes it a collection target.
+2. It is not a function that can be called directly by the yorkie user.
```

Comment:
It would be nice to explain this item in Proposal details along with the reason.